Can not deploy Kubernetes Cluster on vSphere with admin user password containing char '%'

```
.phase1.vSphere.username="Administrator@vsphere.local"
.phase1.vSphere.password="%Admin23"
```

```
[container]:/opt/kubernetes-anywhere> make deploy
util/config_to_json .config > .config.json
make do WHAT=deploy-cluster
make[1]: Entering directory '/opt/kubernetes-anywhere'
( cd "phase1/$(jq -r '.phase1.cloud_provider' .config.json)"; ./do deploy-cluster )
RUNTIME ERROR: Unrecognised conversion type: A
	std.jsonnet:313:21-62	function <parse_conv_type>
	std.jsonnet:326:31-59	thunk <ctype>
	std.jsonnet:328:24-28	object <r>
	std.jsonnet:347:38-40	thunk <i>
	std.jsonnet:347:21-66	function <parse_codes>
	std.jsonnet:347:21-66	function <parse_codes>
	std.jsonnet:351:23-49	thunk <codes>
	std.jsonnet:616:30-34	thunk <codes>
	std.jsonnet:517:32-36	
	std.jsonnet:517:21-37	function <format_codes_arr>
	...
	vSphere.jsonnet:132:21-133	thunk <array_element>
	vSphere.jsonnet:(130:27)-(135:19)	object <anonymous>
	vSphere.jsonnet:(129:32)-(136:17)	object <anonymous>
	vSphere.jsonnet:(128:27)-(137:12)	thunk <array_element>
	vSphere.jsonnet:(128:26)-(141:13)	object <anonymous>
	vSphere.jsonnet:(121:17)-(142:9)	object <anonymous>
	std.jsonnet:(943:13)-(952:13)	object <anonymous>
	std.jsonnet:(943:13)-(952:13)	object <anonymous>
	all.jsonnet:3:49-79	object <anonymous>
	During manifestation	
Makefile:63: recipe for target 'do' failed
make[1]: *** [do] Error 1
make[1]: Leaving directory '/opt/kubernetes-anywhere'
```

cc: @tusharnt, @BaluDontu @abrarshivani

--- a/Makefile
+++ b/Makefile
@@ -87,7 +87,7 @@ docker-push: docker-build
 	docker push $(IMAGE_NAME):$(IMAGE_VERSION)
 
 clean:
-	( if [[ -e "${CONFIG_JSON_FILE}" ]];then rm -rf phase3/${CLOUD_PROVIDER}/.tmp/ phase1/${CLOUD_PROVIDER}/${CLUSTER_NAME}/; fi )
+	(if [[ ! -z "${CLOUD_PROVIDER}" && ! -z "${CLUSTER_NAME}" ]];then rm -rf phase3/${CLOUD_PROVIDER}/.tmp/ phase1/${CLOUD_PROVIDER}/${CLUSTER_NAME}/; fi)
 
 fmt:
 	for f in $$(find . -name '*.jsonnet'); do jsonnet fmt -i -n 2 $${f}; done;

--- a/phase1/vsphere/vSphere.jsonnet
+++ b/phase1/vsphere/vSphere.jsonnet
@@ -15,13 +15,12 @@ function(config)
         "https://${vsphere_virtual_machine.kubevm1.network_interface.0.ipv4_address}",
       ));
 
-  local config_metadata_template = std.toString(config {
+  local config_metadata_template = config {
       master_ip: "${vsphere_virtual_machine.kubevm1.network_interface.0.ipv4_address}",
-      role: "%s",
       phase3 +: {
         addons_config: (import "phase3/all.jsonnet")(config),
       },
-    });
+    };
   
   std.mergePatch({
     // vSphere Configuration
@@ -135,7 +134,7 @@ function(config)
                 "remote-exec": {
                   inline: [
                     "hostnamectl set-hostname %s" % std.join("", [cfg.cluster_name, "-master"]),
-                    "mkdir -p /etc/kubernetes/; echo '%s' > /etc/kubernetes/k8s_config.json " % (config_metadata_template % "master"),                    
+                    "mkdir -p /etc/kubernetes/; echo '%s' > /etc/kubernetes/k8s_config.json " % std.toString((config_metadata_template {role: "master"})),
                     "echo '%s' >  /etc/kubernetes/vsphere.conf" % "${data.template_file.cloudprovider.rendered}",            
                     "echo '%s' > /etc/configure-vm.sh; bash /etc/configure-vm.sh" % "${data.template_file.configure_master.rendered}",
                   ]
@@ -157,7 +156,7 @@ function(config)
                 "remote-exec": {
                   inline: [
                     "hostnamectl set-hostname %s" % std.join("", [cfg.cluster_name, "-node", vm-1]),
-                    "mkdir -p /etc/kubernetes/; echo '%s' > /etc/kubernetes/k8s_config.json " % (config_metadata_template % "node"),                    
+                    "mkdir -p /etc/kubernetes/; echo '%s' > /etc/kubernetes/k8s_config.json " % std.toString((config_metadata_template {role: "node"})),
                     "echo '%s' > /etc/configure-vm.sh; bash /etc/configure-vm.sh" % "${data.template_file.configure_node.rendered}",
                     "echo '%s' >  /etc/kubernetes/vsphere.conf" % "${data.template_file.cloudprovider.rendered}",            
                   ]

